src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1

```diff
@@ -0,0 +1,309 @@
+            {
+                $CompText = "is"
+            }
+            else
+            {
+                $CompText = "is not"
+            }
+
+            return $CompText
+        }
+
+        Function CreatePerfTuningStatus {
+            param(
+                [Parameter(Mandatory = $true)]
+                [String]$Title,
+
+                [Parameter(Mandatory = $true)]
+                [ScriptBlock]$ScriptBlock
+            )
+
+            $PerfTuningStatus = & $ScriptBlock
+
+            $PerfTuningStatus.Title = $Title
+
+            $TextStatus = "WILL"
+            if (-not $PerfTuningStatus.ShouldBeSkipped)
+            {
+                $TextStatus = "$TextStatus DO"
+            } else 
+            {
+                $TextStatus = "$TextStatus SKIP"
+            }
+
+            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message "$($PerfTuningStatus.Title): [$TextStatus]"
+
+            foreach ($Message in $PerfTuningStatus.Messages)
+            {
+                LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message " > $Message"
+            }
+
+            return $PerfTuningStatus
+        }
+
+        Function CreateResult {
+            param(
+                [Parameter(Mandatory = $true)]
+                [Bool]$ShouldBeSkipped,
+
+                [Parameter(Mandatory = $true)]
+                [AllowEmptyCollection()]
+                [String[]]$Messages
+            )
+
+            $Result = @{}
+            $Result.ShouldBeSkipped = $ShouldBeSkipped
+            $Result.Messages = $Messages
+
+            return $Result
+        }
+
+        $PerfTuning = @{}
+        $PerfTuning.SomethingToDo = $False
+        $PerfTuning.Sections = @{}
+    }
+
+    process
+    {
+        # Check if user is admin
+        if (-not $(IsAdmin))
+        {
+            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 3 -Message "The current user is not Administrator or not running this script in an elevated session"
+            WriteNonTerminalError -Message "The current user is not Administrator or not running this script in an elevated session"
+
+            $installResult.Success = $False
+            $installResult.ExitCode = -1
+            $installResult.Message = 'The current user is not Administrator or not running this script in an elevated session'
+
+            return $installResult
+        }
+ 
+        $PerfTuning.Sections.ProcessSchedulingConfig = CreatePerfTuningStatus -Title "Setting Windows processor scheduling priority to 'background services'" `
+            -ScriptBlock `
+            {
+                $ShouldBeSkipped = $True
+                $Messages = @()
+
+                try
+                {
+                    $Value = RegRead -Path "HKLM:\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\PriorityControl" -Name "Win32PrioritySeparation" -Type "Dword"
+
+                    $ShouldBeSkipped = ($Value -eq 24)
```

Comment:
(same as above) We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1